docs/src/main/sphinx/functions/setdigest.rst

```diff
@@ -0,0 +1,156 @@
+=========================
+
+Trino offers several functions that deal with
+`MinHash <https://en.wikipedia.org/wiki/MinHash>`_ technique.
+
+MinHash is a technique  which is employed to quickly estimate the
+`Jaccard similarity coefficient <https://en.wikipedia.org/wiki/Jaccard_index>`_
+between two sets.
+
+One of the common applications of the MinHash technique is in data mining where it is employed for
+detecting at scale near-duplicate web pages. By using this information, the search engines
+efficiently avoid showing two pages which are nearly identical within the search results.
+
+Data structures
+---------------
+
+Trino implements Set Digest data sketches by encapsulating the following components:
+
+- `HyperLogLog <https://en.wikipedia.org/wiki/HyperLogLog>`_
+- `MinHash with a single hash function <http://en.wikipedia.org/wiki/MinHash#Variant_with_a_single_hash_function>`_
+
+The Hyperloglog structure is used for the aproximation of the distinct elements
+in the original set.
+
+The MinHash structure is used to store a low memory footprint signature of the original set.
+The similarity of any two sets is estimated by comparing their signatures.
+
+The Trino type for this data structure is called ``setdigest``.
+
+Trino offers the ability to merge multiple Set Digest data sketches.
+
+The following example showcases how the Set Digest functions are
+employed to estimate the similarity between texts::
+
+    WITH text_input(id, text) AS (VALUES
+                                 (1, 'The quick brown fox jumps over the lazy dog'),
+                                 (2, 'The quick red fox jumps over the lazy dog'),
+                                 (3, 'The quick and the lazy'),
+                                 (4, 'The quick brown fox jumps over the dog')),
+         text_ngrams(id, ngrams) AS (
+             SELECT id,
+                    transform(
+                      ngrams(
+                        split(text, ' '),
+                        4
+                      ),
+                      token -> array_join(token, ' ')
+                    )
+             FROM text_input
+         ),
+         minhash_digest(id, digest) AS (
+             SELECT id,
+                    (SELECT make_set_digest(v) FROM unnest(ngrams) u(v))
+             FROM text_ngrams
+         ),
+         setdigest_side_by_side(id1, digest1, id2, digest2) AS (
+             SELECT m1.id as id1,
+                    m1.digest as digest1,
+                    m2.id as id2,
+                    m2.digest as digest2
+             FROM (SELECT id, digest FROM minhash_digest) m1
+             JOIN (SELECT id, digest FROM minhash_digest) m2
+               ON m1.id != m2.id AND m1.id < m2.id
+        )
+    SELECT id1,
+           id2,
+           intersection_cardinality(digest1, digest2) AS intersection_cardinality,
+           jaccard_index(digest1, digest2)            AS jaccard_index
+    FROM setdigest_side_by_side
+    ORDER BY id1, id2;
+
```

Comment:
Add a sentence here

Reply:
What would be an appropriate sentence in this context?
Are you implying  that the reader may think that the two code blocks are not related?

Reply:
It should be a full sentence. E.g. 

The query produces the following results: